anndata/_core/sparse_dataset.py

```diff
@@ -474,6 +474,15 @@ def append(self, sparse_matrix: ss.spmatrix):
                 f"Matrices must have same format. Currently are "
                 f"{self.format!r} and {get_format(sparse_matrix)!r}"
             )
+        indptr_offset = self.group["indptr"][-1]
+        if indptr_offset.dtype == np.int32:
+            new_nnz = int(indptr_offset) + int(sparse_matrix.indptr[-1])
```

Comment:
Why the casting if these should be ints?

Reply:
`indptr_offset` and the other value are likely int32, and I want to avoid overflow.

```python
typemax_int32 = np.int32(np.iinfo(np.int32).max)
assert (typemax_int32 + np.int32(1)) < 0
```